File: ROUN1B/max2.cpp
#include<iostream>
#include<stdlib.h>
#include<vector>
#define L 1000000007
void check(long f, long K, std::vector<unsigned long long> & Z){
    for (long i= K/2 ;i>=1;i--)
        //std::cout<<(f-i)<<"\n";
        Z.push_back(f-i);
    if (K%2!=0)
        //std::cout<<f<<"\n";
        Z.push_back(f);
    for (long i=1 ;i<=K/2;i++)
        //std::cout<<(f+i)<<"\n";
        Z.push_back(f+i);
}
main(){
    std::ios::sync_with_stdio(0);
    std::cin.tie(NULL);
    int T, temp;
    long N, K, f,R, i;
    unsigned long long out;
    short o;
    std::cin>>T;
    if (T<1||T>1000)
        exit(0);
    for (int t=0;t<T;t++){
        std::cin>>N>>K;
        if(N<1||K<1||N>1000000000||K>10000)
            exit(0);
        temp =  ((K-1)*(2+K))/2; //2..1+(K-1) summation
        out = 1;
        o = 1;
        if (temp> (N-1) || N<K)
            o = -1;
        else if(temp == N-1)
            o = 0;
        else{
            f = N/K;
            R = N%K;
            std::vector<unsigned long long> Z;
            check(f,K,Z);
            if(N%K != 0){//Rem
                if (K%2 ==0){//even
                    i = K/2 - 1;
                    while(R!=0 && i>=0){
                        Z[i] += 1;
                        i--;
                        R--;
                    }
                    i = K-1;
                    while(R!=0){
                        Z[i] += 1;
                        i--;
                        R--;
                    }
                }
                else{//odd
                    i = K-1;
                    while(R!=0){
                        Z[i] += 1;
                        i--;
                        R--;
                    }
                }
            }
            //std::cout<<sizeof(Z)/sizeof(long);
            //long sum = 0;
            for(long a = 0;a<K;a++){
                //f = Z[a]%L;
                //out *= ( ( f * ( (f -1) % L ) ) %L );
                std::cout<<a<<"\t";
                //sum += a;
            }
            //std::cout<<"\n"<<sum<<"\n";
        }
        
        if (o==-1||o==0)
            std::cout<<o<<"\n";
        else
            std::cout<<(out)<<"\n";
    }
}

File: ROUN1B/max2.py
L = 10**9+7
T =0
temp = 0
f = 0
R = 0
i = 0
out = 0
T = int(input())
if (T<1 or T>1000):
        exit(0)
for t in range(T):
    N,K=[int(X) for X in input().split()]
    if(N<1 or K<1 or N>1000000000 or K>10000):
            exit(0)
    temp =  int(((K-1)*(2+K))/2); 
    out = 1
    if temp> (N-1):
        out = -1
    elif temp == N-1:
        out = 0
    else:
        f = int(N/K)
        R = N%K
        Z = []
        if(N%K != 0):
            i = int(K/2)
            while i>=1:
                Z.append(f-i)
                i -= 1
            if (K%2!=0):
                Z.append(f)
            for i in range (1,int(K/2)+1):
                Z.append(f+i)
            if (K%2 ==0):
                i = int(K/2) - 1
                while(R!=0 and i>=0):
                    Z[i] += 1
                    i-=1
                    R-=1
                i = K-1
                while(R!=0):
                    Z[i] += 1
                    i-=1
                    R-=1
            else:
                i = K-1
                while (R!=0):
                    Z[i] += 1
                    i-=1
                    R-=1
            for a in Z:
                print(a)
                f = a%L
                out *= ( ( f * ( (f -1) % L ) ) %L )
        else:
            i = int(K/2)
            while i>=1:
                out *= (((f-i)%L*(f-i-1)%L)%L)
                i -= 1
            if (K%2!=0):
                out *= (((f)%L*(f-1)%L)%L)
            for i in range (1,int(K/2)+1):
                out *= (((f+i)%L*(f+i-1)%L)%L)
                #print(Z)    
    if (out==-1 or out==0):
        print(out)
    else:
        print(out%L)
        